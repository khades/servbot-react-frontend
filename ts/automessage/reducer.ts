import States from "../utils/states";
import { actiontypes, AutoMessageAction } from "./actions";
import { IAutoMessageWithHistory } from "./types";

export interface IAutoMessageState {
    channelID: string;
    id: string;
    isNew: boolean;
    state: States;
    validationError: boolean;
    content?: IAutoMessageWithHistory;
}

const initialState: IAutoMessageState = {
    channelID: "",
    id: "",
    isNew: false,
    state: States.NOTINITIATED,
    validationError: false,
};

const reducer = (state: IAutoMessageState = initialState, action: AutoMessageAction): IAutoMessageState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                id: action.payload.id,
                state: States.LOADING,
            });
        case actiontypes.CREATENEW:
            return {
                channelID: action.payload.channelID,
                content: {
                    channelID: action.payload.channelID,
                    durationLimit: 300000000000,
                    durationThreshold: 0,
                    game: "",
                    history: [],
                    id: "",
                    message: "",
                    messageLimit: 50,
                    messageThreshold: 0,
                },
                id: "",
                isNew: true,
                state: States.READY,
                validationError: false,
            };
        case actiontypes.READY:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                content: action.payload.content,
                id: action.payload.id,
                isNew: false,
                state: States.READY,
                validationError: false,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                id: action.payload.id,
                state: States.NOTAUTHORIZED,
            });
        case actiontypes.NOTFOUND:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                id: action.payload.id,
                state: States.NOTFOUND,
            });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                id: action.payload.id,
                state: States.FORBIDDEN,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, {
                state: States.OFFLINE,
            });
        case actiontypes.SAVE:
            return Object.assign({}, state, {
                state: States.UPDATING,
            });
        case actiontypes.SAVENEW:
            return Object.assign({}, state, {
                state: States.UPDATING,
            });
        case actiontypes.ONSAVEERROR:
            return Object.assign({}, state, {
                state: States.READY,
                validationError: true,
            });
        case actiontypes.RESET:
            return initialState;
        case actiontypes.AFTERCREATION:
            return Object.assign({}, state, { id: action.payload.id });
        default:
            return state;
    }
};

export default reducer;
