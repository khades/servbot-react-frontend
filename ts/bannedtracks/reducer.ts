import States from "../utils/states";
import { actiontypes, BannedTracksAction } from "./actions";
import { IBannedTracks } from "./types";

export interface IBannedTracksState {
    state: States;
    page: number;
    content?: IBannedTracks;
}

const initialState: IBannedTracksState = {
    page: 1,
    state: States.NOTINITIATED,
};

const reducer = (state: IBannedTracksState = initialState, action: BannedTracksAction): IBannedTracksState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return {
                page: 1,
                state: States.LOADING,
            };
        case actiontypes.READY:
            return Object.assign({}, state, {
                content: action.payload.content,
                page: action.payload.page,
                state: States.READY,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, {
                state: States.NOTAUTHORIZED,
            });
        case actiontypes.NOTFOUND:
            return Object.assign({}, state, {
                state: States.NOTFOUND,
            });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                state: States.FORBIDDEN,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, {
                state: States.OFFLINE,
            });
        case actiontypes.UNBAN:
            return Object.assign({}, state, {
                state: States.UPDATING,
            });
        case actiontypes.RESET:
            return initialState;
        default:
            return state;
    }
};

export default reducer;
