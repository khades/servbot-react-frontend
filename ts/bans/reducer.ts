import { IBan } from "../api/types";
import States from "../utils/states";
import { actiontypes, BansAction } from "./actions";

export interface IBansState {
    bans?: IBan[];
    state: States;
    channelID: string;
}

const initialState = {
    channelID: "",
    state: States.NOTINITIATED,
};

const reducer = (state: IBansState = initialState, action: BansAction): IBansState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return Object.assign({}, action.payload, { state: States.LOADING });
        case actiontypes.READY:
            return Object.assign({}, action.payload, { state: States.READY });
        case actiontypes.NOTFOUND:
            return Object.assign({}, action.payload, { state: States.NOTFOUND });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, action.payload, { state: States.FORBIDDEN });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, action.payload, { state: States.NOTAUTHORIZED });
        case actiontypes.OFFLINE:
            return Object.assign({}, action.payload, { state: States.OFFLINE });
        default:
            return state;
    }
};

export default reducer;
