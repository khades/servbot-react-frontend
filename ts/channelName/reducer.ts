import States from "../utils/states";
import { actiontypes, ChannelNameActions } from "./actions";

export interface IChannelName {
    name?: string;
    state: States;
}
export interface IChannelNameStore {
    [key: string]: IChannelName;
}

const reducer = (state: IChannelNameStore = {}, action: ChannelNameActions): IChannelNameStore => {
    switch (action.type) {
        case actiontypes.READY:
            return Object.assign({}, state, {
                [action.payload.channelNameID]: {
                    name: action.payload.channelName,
                    state: States.READY,
                },
            });
        case actiontypes.NOTFOUND:
            return Object.assign({}, state, {
                [action.payload.channelNameID]: {
                    state: States.NOTFOUND,
                },
            });
        case actiontypes.LOADING:
            return Object.assign({}, state, {
                [action.payload.channelNameID]: {
                    state: States.LOADING,
                },
            });
        default:
            return state;
    }
};

export default reducer;
