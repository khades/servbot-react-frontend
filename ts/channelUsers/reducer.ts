import { IUserLogsInfo } from "../userLogs/types";
import States from "../utils/states";
import { actiontypes, ChannelUsersAction } from "./actions";

export interface IChannelUsersState {
    channelID: string;
    userName: string;
    users: IUserLogsInfo[];
    state: States;
}

const initialState: IChannelUsersState = {
    channelID: "",
    state: States.NOTINITIATED,
    userName: "",
    users: [],
};

const reducer = (state: IChannelUsersState = initialState, action: ChannelUsersAction) => {

    switch (action.type) {
        case actiontypes.READY:
            return Object.assign({}, state, {
                state: States.READY,
                users: action.payload,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, { state: States.OFFLINE });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.FORBIDDEN,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, { state: States.NOTAUTHORIZED });
        case actiontypes.LOADING:
            if (state.channelID === action.payload.channelID) {
                return Object.assign({}, state, {
                    channelID: action.payload.channelID,
                    state: States.UPDATING,
                    userName: action.payload.userName,
                });
            } else {
                return {
                    channelID: action.payload.channelID,
                    state: States.LOADING,
                    userName: action.payload.userName,
                    users: [],
                };
            }
        case actiontypes.RESET:
            return initialState;
        default:
            return state;
    }
};

export default reducer;
