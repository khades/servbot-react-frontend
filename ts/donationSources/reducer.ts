import States from "../utils/states";
import { actiontypes, DonationSourcesAction } from "./actions";
import { IDonationSources } from "./types";

export interface IDonationSourcesState {
    channelID: string;
    state: States;
    content?: IDonationSources;
}

const initialState: IDonationSourcesState = {
    channelID: "",
    state: States.NOTINITIATED,
};

const reducer = (state: IDonationSourcesState = initialState, action: DonationSourcesAction): IDonationSourcesState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.LOADING,
            });
        case actiontypes.READY:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                content: action.payload.content,
                state: States.READY,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.NOTAUTHORIZED,
            });
        case actiontypes.NOTFOUND:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.NOTFOUND,
            });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.FORBIDDEN,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, {
                state: States.OFFLINE,
            });
        default:
            return state;
    }
};

export default reducer;
