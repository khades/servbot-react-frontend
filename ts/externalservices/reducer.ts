import States from "../utils/states";
import { actiontypes, ExternalServicesAction } from "./actions";
import { IChannelExternalServices } from "./types";

export interface IExternalServicesState {
    channelID: string;
    state: States;
    content?: IChannelExternalServices;

}

const initialState: IExternalServicesState = {
    channelID: "",
    state: States.NOTINITIATED,
};

const reducer = (
    state: IExternalServicesState = initialState,
    action: ExternalServicesAction): IExternalServicesState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.LOADING,
            });
        case actiontypes.READY:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                content: action.payload.content,
                state: States.READY,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.NOTAUTHORIZED,
            });
        case actiontypes.NOTFOUND:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.NOTFOUND,
            });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.FORBIDDEN,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, {
                state: States.OFFLINE,
            });
        case actiontypes.SAVE:
            return Object.assign({}, state, {
                state: States.UPDATING,
            });
        case actiontypes.RESET:
            return initialState;
        default:
            return state;
    }
};

export default reducer;
