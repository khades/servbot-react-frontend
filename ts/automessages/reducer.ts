import { IAutoMessageWithHistory } from "../automessage/types";
import States from "../utils/states";
import { actiontypes, AutoMessagesAction } from "./actions";

export interface IAutoMessagesState {
    channelID: string;
    state: States;
    emptyShown: boolean;
    content?: IAutoMessageWithHistory[];
}

const initialState: IAutoMessagesState = {
    channelID: "",
    emptyShown: false,
    state: States.NOTINITIATED,
};

const reducer = (state: IAutoMessagesState = initialState, action: AutoMessagesAction): IAutoMessagesState => {
    switch (action.type) {
        case actiontypes.LOADING:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.LOADING,
            });
        case actiontypes.READY:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                content: action.payload.content,
                emptyShown: false,
                state: States.READY,
            });
        case actiontypes.NOTAUTHORIZED:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.NOTAUTHORIZED,
            });
        case actiontypes.NOTFOUND:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.NOTFOUND,
            });
        case actiontypes.FORBIDDEN:
            return Object.assign({}, state, {
                channelID: action.payload.channelID,
                state: States.FORBIDDEN,
            });
        case actiontypes.OFFLINE:
            return Object.assign({}, state, {
                state: States.OFFLINE,
            });
        case actiontypes.SHOWEMPTY:
            return Object.assign({}, state, {
                emptyShown: true,
            });
        case actiontypes.HIDEEMPTY:
            return Object.assign({}, state, {
                emptyShown: false,
            });
        default:
            return state;
    }
};

export default reducer;
