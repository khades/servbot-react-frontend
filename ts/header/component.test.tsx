import * as React from "react";
import { create } from "react-test-renderer";
import { SideMenuStates } from "../sidemenu/reducer";
import Header, { IHeaderProps } from "./component";

describe("Header", () => {
    it("should render properly for SideMenuState == SHOWN ", () => {
        const props: IHeaderProps = {
            getSideMenuState: () => SideMenuStates.SHOWN,
            hideMenu: () => 1,
            showMenu: () => 1,
        };
        const header = create(
            <Header {...props} />,
        );
        expect(header.toJSON()).toMatchSnapshot();
    });
    it("should render properly for SideMenuState == HIDDEN ", () => {
        const props: IHeaderProps = {
            getSideMenuState: () => SideMenuStates.HIDDEN,
            hideMenu: () => 1,
            showMenu: () => 1,
        };
        const header = create(
            <Header {...props} />,
        );
        expect(header.toJSON()).toMatchSnapshot();
    });
    it("should render properly for SideMenuState == INIT ", () => {
        const props: IHeaderProps = {
            getSideMenuState: () => SideMenuStates.INIT,
            hideMenu: () => 1,
            showMenu: () => 1,
        };
        const header = create(
            <Header {...props} />,
        );
        expect(header.toJSON()).toMatchSnapshot();
    });
});
