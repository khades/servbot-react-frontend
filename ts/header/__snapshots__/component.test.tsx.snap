// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly for SideMenuState == HIDDEN  1`] = `
<div
  className="headerContent"
>
  <div
    className="headerContent__menu-button-container"
    onClick={[Function]}
  >
    <div
      className="headerContent__menu-button"
    />
  </div>
</div>
`;

exports[`Header should render properly for SideMenuState == INIT  1`] = `
<div
  className="headerContent"
>
  <div
    className="headerContent__menu-button-container"
    onClick={[Function]}
  >
    <div
      className="headerContent__menu-button"
    />
  </div>
</div>
`;

exports[`Header should render properly for SideMenuState == SHOWN  1`] = `
<div
  className="headerContent"
>
  <div
    className="headerContent__menu-button-container headerContent__menu-button__menu-shown"
    onClick={[Function]}
  >
    <div
      className="headerContent__menu-button"
    />
  </div>
</div>
`;
